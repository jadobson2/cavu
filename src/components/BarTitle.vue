<template>
  <header class="
    sticky top-0 flex items-center md:mb-3 px-8 xl:px-14 py-6 lg:py-8
    bg-radial-gradient-yellow"
  >
    <slot name="icon"></slot>
    <h1 class="text-4xl lg:text-5xl font-bold not-first:ml-8">{{ title }}</h1>
  </header>
</template>

<script lang="ts">
import Vue from 'vue'

export default Vue.extend({
  name: 'BarTitle',
  props: {
    title: String
  }
})
</script>
