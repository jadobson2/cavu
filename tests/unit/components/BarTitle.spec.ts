import { shallowMount } from '@vue/test-utils'
import BarTitle from '@/components/BarTitle.vue'

describe('BarTitle', () => {
  it('renders correctly', () => {
    const wrapper = shallowMount(BarTitle, {
      propsData: { title: 'Title' },
      slots: { icon: '<img src="foo.svg"/>' }
    })

    expect(wrapper.element).toMatchSnapshot()
  })
})
