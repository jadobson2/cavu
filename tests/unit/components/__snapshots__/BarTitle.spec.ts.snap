// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BarTitle renders correctly 1`] = `
<header
  class="sticky top-0 flex items-center md:mb-3 px-8 xl:px-14 py-6 lg:py-8 bg-radial-gradient-yellow"
>
  <img
    src="foo.svg"
  />
   
  <h1
    class="text-4xl lg:text-5xl font-bold not-first:ml-8"
  >
    Title
  </h1>
</header>
`;
